src/linker/Linker/MethodReferenceExtensions.cs

```diff
@@ -56,15 +56,15 @@ public static string GetDisplayName (this MethodReference method)
 		public static TypeReference GetReturnType (this MethodReference method, LinkContext context)
 		{
 			if (method.DeclaringType is GenericInstanceType genericInstance)
-				return TypeReferenceExtensions.InflateGenericType (genericInstance, method.ReturnType, context);
+				return TypeReferenceExtensions.InflateGenericType (genericInstance, method.ReturnType, context)!;
```

Comment:
Is there a reason `InflateGenericType` can return null? (it seems to handle null input, but that can be changed ... and it would make sense)

Reply:
The only way I see it not working is when is not able to resolve the GenericInstanceProvider.ElementType, does a GenericInstanceType always have a ElementType?